src/language/typescript.ts

```diff
@@ -671,21 +673,65 @@ const client = `
 		} 
 	
 		constructor(readonly errors: Errors) {
-			super('ResponseValidationError');
+			super(report(left(errors)).toString());
+			this.name = 'ResponseValidationError';
 			Object.setPrototypeOf(this, ResponseValidationError);
 		}
 	}
 `;
 
 const utils = `
-	import { Type, success, identity } from 'io-ts';
+	import { Type, success, identity, Errors, Validation, Context, ValidationError } from 'io-ts';
+
 	export const unknownType = new class UnknownType extends Type<unknown> {
 		readonly _tag: 'UnknownType' = 'UnknownType';
-	
+
 		constructor() {
 			super('unknownType', (_: unknown): _ is unknown => true, success, identity);
 		}
 	}();
+
+	function getMessage(e: ValidationError) {
+		return e.message !== undefined ? e.message : 
+			createMessage(e.context)
+			+ '\\n in context: \\n' 
+			+ getContextPath(e.context);
+	}
+
+	function createMessage(context: Context) {
+		return (
+			'\\n Received: \\n  ' +
+			JSON.stringify(context[context.length - 1].actual) +
+			'\\n expected: \\n  ' +
+			context[context.length - 1].type.name +
+			'\\n in field \\n  ' +
+			context[context.length - 1].key
+		);
+	}
+
+	function getContextPath(context: Context) {
+		return context
+			.map(function(cEntry, index) {
+				const padding = new Array(index * 2 + 2).fill(' ').join('');
```

Comment:
`String.prototype.padLeft`?

Reply:
There exists padStart, but id doesn't seem to fit here, since i dont't know the resulting string's length in advance